lib/features/modeling/cmd/CreateElementsHandler.js

```diff
@@ -0,0 +1,118 @@
+import {
+  assign,
+  forEach,
+  isNumber,
+  map,
+  pick,
+  values
+} from 'min-dash';
+
+import {
+  getBBox,
+  getParents
+} from '../../../util/Elements';
+
+
+export default function CreateElementsHandler(modeling) {
+  this._modeling = modeling;
+}
+
+CreateElementsHandler.$inject = [
+  'modeling'
+];
+
+CreateElementsHandler.prototype.preExecute = function(context) {
+  var elements = context.elements,
+      parent = context.parent,
+      parentIndex = context.parentIndex,
+      position = context.position,
+      hints = context.hints;
+
+  var modeling = this._modeling;
+
+  // make sure each element has x and y
+  forEach(elements, function(element) {
+    if (!isNumber(element.x)) {
+      element.x = 0;
+    }
+
+    if (!isNumber(element.y)) {
+      element.y = 0;
+    }
+  });
+
+  var bbox = getBBox(elements);
+
+  // center elements around position
+  forEach(elements, function(element) {
+    if (isConnection(element)) {
+      element.waypoints = map(element.waypoints, function(waypoint) {
+        return {
+          x: waypoint.x - bbox.x - bbox.width / 2 + position.x,
+          y: waypoint.y - bbox.y - bbox.height / 2 + position.y
+        };
+      });
+    }
+
+    assign(element, {
+      x: element.x - bbox.x - bbox.width / 2 + position.x,
+      y: element.y - bbox.y - bbox.height / 2 + position.y
+    });
+  });
+
+  var parents = getParents(elements);
+
+  var cache = {};
+
+  forEach(elements, function(element) {
+    if (isConnection(element)) {
+      cache[ element.id ] = isNumber(parentIndex) ?

```

Comment:
The whole declaration looks very huge. Can we somehow make it a little shorter? I know it's hard because of the param-order.

Maybe someday we can refactor the used function to use objects...

Reply:
Right now we have to work with the existing `Modeling#createShape` and `Modeling#createConnection` methods. In general, I think the modeling API methods are not supposed to only have one parameter since they abstract calling a command handler which then only takes one parameter.